cmd/state/errors.go

```diff
@@ -5,17 +5,77 @@ import (
 	"fmt"
 	"os"
 	"runtime/debug"
+	"strings"
 	"time"
 
+	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/failures"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/output/txtstyle"
 	"github.com/ActiveState/cli/internal/rtutils"
 	"github.com/ActiveState/cli/internal/subshell/sscommon"
 )
 
+type ErrorTips interface {
+	ErrorTips() []string
+}
+
+type OutputError struct {
+	error
+}
+
+func (o *OutputError) MarshalOutput(f output.Format) interface{} {
+	if f != output.PlainFormatName {
+		return o.error
+	}
+
+	// Print header
+	errs := locale.UnwrapError(o.error)
+	title := txtstyle.NewTitle(trimError(locale.ErrorMessage(o.error)))
+	title.Heading = locale.Tl("err_heading", " ERROR! ")
+	title.ColorCode = "ERROR"
+
+	var outLines []string
+	outLines = append(outLines, title.String())
+
+	// Print what happened
+	if len(errs) > 1 {
+		outLines = append(outLines, output.Heading(locale.Tl("err_what_happened", "Here's What Happened")).String())
+
+		for _, errv := range errs[1:] {
+			outLines = append(outLines, fmt.Sprintf(" [ERROR]x[/RESET] %s", trimError(locale.ErrorMessage(errv))))
+		}
+	}
+
+	// Concatenate error tips
+	errorTips := []string{}
+	err := o.error
+	for err != nil {
+		if v, ok := err.(ErrorTips); ok {
+			errorTips = append(errorTips, v.ErrorTips()...)
+		}
+		err = errors.Unwrap(err)
+	}
+	errorTips = append(errorTips, locale.Tl("err_help_forum", "Community → [ACTIONABLE]{{.V0}}[/RESET]", constants.ForumsURL))
+
+	// Print tips
+	outLines = append(outLines, output.Heading(locale.Tl("err_more_help", "Need More Help?")).String())
+	for _, tip := range errorTips {
+		outLines = append(outLines, fmt.Sprintf(" [DISABLED]-[/RESET] %s", trimError(tip)))
+	}
+	return "\n" + strings.Join(outLines, "\n")
+}
+
+func trimError(msg string) string {
+	if strings.Count(msg, ".") > 1 || strings.Count(msg, ",") > 1 {
+		return msg // Don't trim dots if we have multiple sentences.
+	}
+	return strings.TrimRight(msg, " .")
+}
```

Comment:
I don't quite understand what the `trimError` function is supposed to do. 
Could you please explain?

 I also noticed that it is currently skipping the trimming when there are at least *2* dots or commas in the message.   Should it be *1* instead?

Reply:
This is making the errors consistently not have a period at or whitespace at the end UNLESS the sentence includes a comma or multiple dots (ie. is more than one sentence).

Reply:
Should it then be: ``` strings.Count(msg, ",") > 0 ``` ?

Reply:
Yes, you're right! fixed.